.github/workflows/publish.yml

```diff
@@ -3,16 +3,17 @@ name: publish
 on:
   push:
     branches: ['main']
+    tags: ['v*.*.*']
```

Comment:
Not blocking , but this may be catching `.*` as regex instead of implementing a check for actual periods.  It probably doesn't matter but it may be better as `v[0-9]+\.[0-9]+\.[0-9]+`

Reply:
Honestly, v.* would be fine here.